src/main/java/com/redhat/pantheon/servlet/AbstractJsonSingleQueryServlet.java

```diff
@@ -0,0 +1,69 @@
+package com.redhat.pantheon.servlet;
+
+import com.redhat.pantheon.jcr.JcrQueryHelper;
+import com.redhat.pantheon.model.QuerySingleResult;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nonnull;
+import javax.jcr.RepositoryException;
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.google.common.collect.Maps.newHashMap;
+import static com.redhat.pantheon.servlet.ServletUtils.writeAsJson;
+
+/**
+ * A base servlet that helps to create simple Json result from custom queries.
+ * This servlet should be extended to create components that render a single element of
+ * results in json format. See methods to be overriden for details on how to
+ * customize sub classes.
+ *
+ * @author Ankit Gadgil
+ */
+public abstract class AbstractJsonSingleQueryServlet extends SlingSafeMethodsServlet {
+
+    /**
+     * Returns the query to execute. The query may be modified depending on the provided
+     * parameters in the request.
+     * @param request The sling servlet request
+     * @return A string with the query to execute. The results of this query will be used
+     * to render the list of results.
+     */
+    protected abstract String getQuery(SlingHttpServletRequest request);
+
+    /**
+     * Provides a way to modify the returned objects based on the found resources.
+     * The default implementation just returns the corresponding value map.
+     * @param resource The Resource obtained as a result of the query.
+     * @return A map with the actual value to be returned to the servlet's caller.
+     */
+    protected Map<String, Object> resourceToMap(@NotNull Resource resource) {
+        return newHashMap(resource.getValueMap());
+    }
+
+    @Override
+    protected final void doGet(@Nonnull SlingHttpServletRequest request, @Nonnull SlingHttpServletResponse response)
+            throws ServletException, IOException {
+
+        JcrQueryHelper queryHelper = new JcrQueryHelper(request.getResourceResolver());
+        try {
+            Stream<Resource> resultStream = queryHelper.query(getQuery(request));
+            List<Map<String, Object>> resultList = resultStream.map(this::resourceToMap)
+                    .collect(Collectors.toList());
+            QuerySingleResult queryResult = new QuerySingleResult(resultList.get(0));
```

Comment:
Probably need to check there is at least one result, otherwise this will throw an exception. 
And if there are no results found, this should probably return a 404 to the caller.